sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1462,13 +1480,50 @@ export class BlobClient extends StorageClient {
    */
   public async downloadToBuffer(
     buffer: Buffer,
-    offset: number,
+    offset?: number,
     count?: number,
-    options: BlobDownloadToBufferOptions = {}
-  ): Promise<Buffer> {
+    options?: BlobDownloadToBufferOptions
+  ): Promise<Buffer>;
+
+  public async downloadToBuffer(
+    param1: Buffer | number = 0,
+    param2: number = 0,
+    param3: BlobDownloadToBufferOptions | number = 0,
+    param4: BlobDownloadToBufferOptions = {}
+  ) {
+    let buffer: Buffer;
+    let offset: number;
+    let count: number;
+    let options = param1 instanceof Buffer ? param4 : (param3 as BlobDownloadToBufferOptions);
     const { span, spanOptions } = createSpan("BlobClient-downloadToBuffer", options.spanOptions);
 
     try {
+      if (param1 instanceof Buffer) {
+        buffer = param1;
+        offset = param2;
+        count = param3 as number;
+      } else {
+        offset = param1;
+        count = param2 as number;
```

Comment:
if `count` is not passed in, i.e., undefined, please ensure we don't make another `getProperties()` call down further.

Reply:
Done. Only one `getProperties` call is being made now.